### ISSUE TITLE
New Rule: `require-suggestions-assertion`

### ISSUE THREAD

**Issue by bmish** [not-an-aardvark/eslint-plugin-eslint-plugin#189]
It would require each invalid test case to assert what suggestions are provided for each violation (using `suggestions: []` if the violation provides no assertions). 

Why?
* Currently, it's easy to forget to assert what suggestions a test case yields, or that a test case yields no assertions. As a result, this can be an area of poor test coverage.

```js
new RuleTester().run('example-rule', rule, {
  valid: [],
  invalid: [
    {
      code: `foo();`,
      output: null,
      errors: [
        {
          messageId: 'bar',
          type: 'Property',
          suggestions: [], // or array of suggestions if rule offers them
        },
      ],
    },
  ]
});
```

Optionally, we could add a `consistent` option to this rule, which would only require the `suggestions` assertion if at least one test case in the file has it (similar to [eslint-plugin/consistent-output](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/docs/rules/consistent-output.md)).

I hope that a future major version of ESLint will require test cases that yield suggestions to assert what they are, so that this lint rule wouldn't be as necessary as it is now.

**Comment by bmish** [comment]
Closing because I plan to pursue an improvement in ESLint v9 (https://github.com/eslint/eslint/issues/15104) that will result in this rule being no longer needed.
